diff --git a/build/android/gyp/compile_java.py b/build/android/gyp/compile_java.py
--- a/build/android/gyp/compile_java.py
+++ b/build/android/gyp/compile_java.py
@@ -604,7 +604,7 @@ def main(argv,
       # Jacoco does not currently support a higher value.
       '--release',
       # Keep consistent with javac_cmd in //build/android/gyp/turbine.py.
-      '21',
+      '8',
       # Chromium only allows UTF8 source files.  Being explicit avoids
       # javac pulling a default encoding from the user's environment.
       '-encoding',
diff --git a/build/android/gyp/turbine.py b/build/android/gyp/turbine.py
--- a/build/android/gyp/turbine.py
+++ b/build/android/gyp/turbine.py
@@ -91,7 +91,7 @@ def main(argv):
   # Keep consistent with javac_args release flag in
   # //build/android/gyp/compile_java.py and
   # compileOptions in //build/android/gradle/android.jinja
-  javac_cmd = ['--release', '21']
+  javac_cmd = ['--release', '8']
 
   # Turbine reads lists from command line args by consuming args until one
   # starts with double dash (--). Thus command line args should be grouped
diff --git a/sdk/android/api/org/webrtc/EglRenderer.java b/sdk/android/api/org/webrtc/EglRenderer.java
--- a/sdk/android/api/org/webrtc/EglRenderer.java
+++ b/sdk/android/api/org/webrtc/EglRenderer.java
@@ -653,7 +653,7 @@ public class EglRenderer implements VideoSink {
               eglBase.swapBuffers();
             }
 
-            for (var listener : renderListeners) {
+            for (RenderListener listener : renderListeners) {
               listener.onRender(System.nanoTime());
             }
 
diff --git a/sdk/android/api/org/webrtc/NetworkMonitorAutoDetect.java b/sdk/android/api/org/webrtc/NetworkMonitorAutoDetect.java
--- a/sdk/android/api/org/webrtc/NetworkMonitorAutoDetect.java
+++ b/sdk/android/api/org/webrtc/NetworkMonitorAutoDetect.java
@@ -781,7 +781,8 @@ public class NetworkMonitorAutoDetect extends BroadcastReceiver implements Netwo
           fieldTrialsString, "unifiedCommunications", /* defaultValue= */ false)) {
         unifiedCommunicationsCb = new SimpleNetworkCallback("slice", availableNetworks);
         try {
-          var request = connectivityManagerDelegate.createUnifiedCommunicationsNetworkRequest();
+          NetworkRequest request =
+              connectivityManagerDelegate.createUnifiedCommunicationsNetworkRequest();
           connectivityManagerDelegate.registerNetworkCallback(request, unifiedCommunicationsCb);
         } catch (Exception ex) {
           Logging.e(TAG, "Failed to registerNetworkCallback for unifiedCommunications");
diff --git a/third_party/jni_zero/java/src/org/jni_zero/CommonApis.java b/third_party/jni_zero/java/src/org/jni_zero/CommonApis.java
--- a/third_party/jni_zero/java/src/org/jni_zero/CommonApis.java
+++ b/third_party/jni_zero/java/src/org/jni_zero/CommonApis.java
@@ -13,7 +13,7 @@ public class CommonApis {
     private static Object[] mapToArray(Map<Object, Object> map) {
         Object[] ret = new Object[map.size() * 2];
         int i = 0;
-        for (var entry : map.entrySet()) {
+        for (Map.Entry<Object, Object> entry : map.entrySet()) {
             ret[i++] = entry.getKey();
             ret[i++] = entry.getValue();
         }
